apps/src/aichat/views/UserChatMessageEditor.tsx

```diff
@@ -16,14 +19,39 @@ const UserChatMessageEditor: React.FunctionComponent = () => {
     (state: {aichat: AichatState}) => state.aichat.isWaitingForChatResponse
   );
 
-  // TODO: If systemPrompt is undefined, handle this error case.
+  const userId: number = useSelector(
+    (state: {currentUser: CurrentUserState}) => state.currentUser.userId
+  );
+
+  const {currentLevelId, scriptId} = useSelector(
+    (state: {progress: ProgressState}) => state.progress
+  );
+
+  const channelId = getStandaloneProjectId();
+
   const dispatch = useAppDispatch();
+
   const handleSubmit = useCallback(() => {
     if (!isWaitingForChatResponse) {
-      dispatch(submitChatMessage(userMessage));
+      const chatContents: ChatContext = {
+        userMessage,
+        userId,
+        currentLevelId,
+        scriptId,
+        channelId,
+      };
+      dispatch(submitChatContents(chatContents));
```

Comment:
Since you're using a thunk action to send this message to the API, it might be easier to grab all the other redux-driven values in the thunk directly (userId, currentLevelId, etc), since you have access to the entire state object. That way this `useCallback` also has way fewer dependencies

Reply:
Yes - that's much simpler - thanks!